plugins/GSdx/GSTextureCache.cpp

```diff
@@ -2046,30 +2045,39 @@ void GSTextureCache::SourceMap::RemoveAt(Source* s)
 	delete s;
 }
 
-// Query the PaletteMap for a valid Palette, then assign both palette texture pointer and clut copy pointer to the Source object
-void GSTextureCache::AttachPaletteToSource(Source* s, uint16 pal)
+// Query the PaletteMap for a valid Palette, then both its reference, clut copy pointer and eventually palette texture pointer to the Source object
+void GSTextureCache::AttachPaletteToSource(Source* s, uint16 pal, bool need_gs_texture)
 {
-	std::shared_ptr<Palette> p = m_palette_map.LookupPalette(pal);
+	std::shared_ptr<Palette> p = m_palette_map.LookupPalette(pal, need_gs_texture);
 	s->m_palette_obj = p;
-	s->m_palette = p->GetPaletteGSTexture();
+	if (need_gs_texture) {
+		s->m_palette = p->GetPaletteGSTexture();
+	}
 	s->m_clut = p->GetClut();
 }
 
 // GSTextureCache::Palette
 
-// Creates a new palette texture with current clut content, keeping a reference to its copy
-GSTextureCache::Palette::Palette(const GSRenderer* renderer, uint16 pal) {
+// Creates a copy of the current clut with eventually a new palette texture with its content
+GSTextureCache::Palette::Palette(const GSRenderer* renderer, uint16 pal, bool need_gs_texture) {
 	uint16 palette_size = pal * sizeof(uint32);
 	m_clut = (uint32*)_aligned_malloc(palette_size, 64);
 	m_renderer = renderer;
 	memcpy(m_clut, (const uint32*)m_renderer->m_mem.m_clut, palette_size);
-	m_tex_palette = m_renderer->m_dev->CreateTexture(256, 1);
-	m_tex_palette->Update(GSVector4i(0, 0, pal, 1), m_clut, palette_size);
+	if (need_gs_texture) {
+		m_tex_palette = m_renderer->m_dev->CreateTexture(256, 1);
+		m_tex_palette->Update(GSVector4i(0, 0, pal, 1), m_clut, palette_size);
+	}
+	else {
+		m_tex_palette = nullptr;
+	}
 }
 
-// Default destructor, recycles palette texture and frees clut copy
+// Default destructor, frees clut copy and recycles eventual palette texture 
```

Comment:
Would have been nice to fix this too...

Reply:
Didn't know, I guess because nobody pointed them out. I'll make another quick push along with the shared pointer suggestion soon-ish.

Reply:
So did no one reviewing the code actually read the comments before it was merged? The comment issue should have been spotted.

Also, maybe you should PR the changes to potentially avoid quick push after quick push after quick push.

Reply:
Nope, didn't have time. I believe kojin and beard tested it when I asked on discord for tests.

I don't have time to do benchmarks so I'll pass.

As for the comments themselves I'm fine with the purge.

Reply:
@lightningterror please, just go ahead and purge the comments; otherwise I'll open PR this WE :P
As for the benchmarks, I would suggest as positive test cases those that were improved by paltex optimization, e.g. ZoE 2 bosses, MGS 3, Jak 2 (especially "underport level" IIRC), maybe R&C.

Reply:
I think I'll leave the purge to you, I have my hands full with other stuff as it is.